### ISSUE TITLE
Sign-up not working: Pre-flight CORS

### ISSUE THREAD

**Issue by velsietis** [Azure/api-management-developer-portal#1054]
## Bug description

Users are unable to sign up through our self hosted portal because of a CORS error when Captcha is enabled.
The pre-flight CORS response is missing the `Access-Control-Allow-Headers` header.

Error occurs during execution of [BackendService.sendSignupRequest](https://github.com/Azure/api-management-developer-portal/blob/da0e8938a297e92461c18890f3d00d4c4c30d7e9/src/services/backendService.ts#L37) and results in the following message in Edge and Chrome developer console:

```
Access to XMLHttpRequest at 'https://<my-apim-service-name>.developer.azure-api.net/signup' from origin 'https://<my-self-hosted-portal-url>' has been blocked by CORS policy: 
Request header field content-type is not allowed by Access-Control-Allow-Headers in preflight response.
```

> NOTE: We are just using the basic e-mail/password authentication option

#### Request
```http
OPTIONS /signup HTTP/1.1
Host: <my-apim-service-name>.developer.azure-api.net
Connection: keep-alive
Accept: */*
Access-Control-Request-Method: POST
Access-Control-Request-Headers: content-type
Origin: <my-self-hosted-portal-url>
Sec-Fetch-Mode: cors
Sec-Fetch-Site: cross-site
Sec-Fetch-Dest: empty
Referer: https://<my-self-hosted-portal-url>/signup
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.121 Safari/537.36 Edg/85.0.564.63
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9,lt;q=0.8,cy;q=0.7
```
#### Response
```http
HTTP/1.1 200 OK
Transfer-Encoding: chunked
Content-Type: text/html
Content-Encoding: gzip
Vary: Accept-Encoding
X-Powered-By: Express
X-Content-Type-Options: nosniff
X-Frame-Options: sameorigin
X-XSS-Protection: 1; mode=block
Access-Control-Allow-Origin: *
Strict-Transport-Security: max-age=31536000;includeSubDomains;
Date: Tue, 24 Nov 2020 10:33:10 GMT
```

**Comment by velsietis** [comment]
Update: Even with Captcha disabled we get another CORS error when verifying the e-mail, this time with a missing `Access-Control-Allow-Origin` when [UserService.activateUser](https://github.com/Azure/api-management-developer-portal/blob/da0e8938a297e92461c18890f3d00d4c4c30d7e9/src/services/usersService.ts#L74) is called

```
Access to XMLHttpRequest at 'https://<my-apim-service-name>.management.azure-api.net/subscriptions/00000/resourceGroups/00000/providers/Microsoft.ApiManagement/service/<my-apim-service-name>/users/5fbcf3051736ea1b5cb37ad6/identities/Basic/john.doe@examplel.com?appType=developerPortal&api-version=2019-12-01' from origin 'https://<my-self-hosted-portal-url>' has been blocked by CORS policy: 
No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

**Comment by velsietis** [comment]
Closing this issue:
 - My first point is actually explained in the Wiki page on [Enable Re-Captcha][1]
 - My second point has been already been addressed in #1034 (I searched before logging, but forgot to remove the _issopen_ filter!)


[1]:https://github.com/Azure/api-management-developer-portal/wiki/Enable-CAPTCHA
